src/net/metaplaylist/index.ts

```diff
@@ -0,0 +1,358 @@
+          const { period, init, segment } = args;
+          const transport = getTransportPipelinesFromSegment(segment, transports);
+          const segmentParser = transport.video.parser;
+          const offset = (period.start || 0) *
+            (init ? (init.timescale || segment.timescale) :  segment.timescale);
+
+          const parserArgs = getParserBaseArguments(segment, args, offset);
+
+          return segmentParser(parserArgs)
+            .pipe(map(({ segmentData, segmentInfos }) => {
+              if (segmentData == null) {
+                return { segmentData: null, segmentInfos: null, segmentOffset: 0 };
+              }
+              const responseData = segmentData instanceof Uint8Array ?
+                segmentData : new Uint8Array(segmentData);
+
+              if (segmentInfos && segmentInfos.time > -1) {
+                segmentInfos.time += offset;
+              }
+              return {
+                segmentData: responseData,
+                segmentInfos,
+                segmentOffset: offset / segment.timescale,
+              };
+            }));
+        },
+      };
+
+    const textTrackPipeline = {
+      loader: (args: ISegmentLoaderArguments) => {
+        const transport = getTransportPipelinesFromSegment(args.segment, transports);
+        const offset = args.period.start;
+        const parserArgs = getParserBaseArguments(args.segment, args, offset);
+        return transport.text.loader(parserArgs);
+      },
+      parser: (args: ISegmentParserArguments<ArrayBuffer|string|Uint8Array|null>) => {
+        const transport = getTransportPipelinesFromSegment(args.segment, transports);
+        const offset = args.period.start;
+        const parserArgs = getParserBaseArguments(args.segment, args, offset);
+        return transport.text.parser(parserArgs)
+          .pipe(map(({ segmentInfos, segmentData }) => {
+            if (segmentInfos && segmentInfos.time > -1) {
+              segmentInfos.time += offset;
+            }
+            return { segmentData, segmentInfos, segmentOffset: offset };
+          }));
+      },
+    };
+
+    const imageTrackPipeline = {
+      loader(args : ISegmentLoaderArguments) :
+        ILoaderObservable<ArrayBuffer|Uint8Array|null> {
+        const transport = getTransportPipelinesFromSegment(args.segment, transports);
+        const offset = args.period.start;
+        const parserArgs = getParserBaseArguments(args.segment, args, offset);
+        return transport.image.loader(parserArgs);
+      },
+      parser(
+        args : ISegmentParserArguments<ArrayBuffer|Uint8Array|null>
+      ) : ImageParserObservable {
+        const transport = getTransportPipelinesFromSegment(args.segment, transports);
+        const offset = args.period.start;
+        const parserArgs = getParserBaseArguments(args.segment, args, offset);
+        return transport.image.parser(parserArgs)
+          .pipe(map(({ segmentInfos, segmentData }) => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 318. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=367&issues=AWbuDiQgTJjFgJLTjDDy&open=AWbuDiQgTJjFgJLTjDDy)

[ISSUE_KEY]: AWbuDiQgTJjFgJLTjDDy